spec/api_spec.rb

```diff
@@ -146,37 +146,137 @@
+          
+          stub_do_not_mail_list_delete(email: new_email)
+          stub_do_not_mail_list_create(email: old_email)
+        end
 
-      before do
-        stub_find_maropost_contact(email: old_email, status: 404, body: '{"message": "Contact is not present!"}')
+        it 'applies old maropost contact settings to existing maropost contact' do
+          updated_contact = subject
+          lists = updated_contact.lists
+
+          merged_contact = JSON.parse merged_contact_maropost_response
+
+          expect(updated_contact.id).to eq merged_contact['id']
+          expect(updated_contact.email).to eq new_email
+          expect(updated_contact.phone_number).to eq merged_contact['phone']
+          expect(updated_contact.cell_phone_number).to eq merged_contact['cell_phone_number']
+          expect(updated_contact.allow_emails).to be_truthy
+
+          expect(lists[:ama_rewards]).to eq merged_contact['ama_rewards']
+          expect(lists[:ama_membership]).to eq merged_contact['ama_membership']
+          expect(lists[:ama_insurance]).to eq merged_contact['ama_insurance']
+          expect(lists[:ama_travel]).to eq merged_contact['ama_travel']
+          expect(lists[:ama_new_member_series]).to eq merged_contact['ama_new_member_series']
+          expect(lists[:ama_fleet_safety]).to eq merged_contact['ama_fleet_safety']
+          expect(lists[:ovrr_personal]).to eq merged_contact['ovrr_personal']
+          expect(lists[:ovrr_business]).to eq merged_contact['ovrr_business']
+          expect(lists[:ovrr_associate]).to eq merged_contact['ovrr_associate']
+          expect(lists[:ama_vr_reminder]).to eq merged_contact['ama_vr_reminder']
+          expect(lists[:ama_vr_reminder_email]).to eq merged_contact['ama_vr_reminder_email']
+          expect(lists[:ama_vr_reminder_sms]).to eq merged_contact['ama_vr_reminder_sms']
+          expect(lists[:ama_vr_reminder_autocall]).to eq merged_contact['ama_vr_reminder_autocall']
+        end
       end
 
-      it 'returns nil' do
-        expect(subject).to be_nil
+      context 'maropost contact does not exist with old email' do
+        let(:existing_contact_maropost_response) { read_fixture('contacts', 'other_existing_contact.json') }
+
+        let(:old_email) { 'test+001@test.com' }
+        let(:new_email) { 'test@example.com' }
+
+        before do
+          stub_find_maropost_contact(email: old_email, status: 404, body: '{"message": "Contact is not present!"}')
+          stub_find_maropost_contact(email: new_email, status: 200, body: existing_contact_maropost_response)
+
+          stub_do_not_mail_list_exists(email: new_email, status: 404)
+
+          stub_update_maropost_contact(contact_id: 888_000_000, status: 200, body: existing_contact_maropost_response)
+
+          stub_do_not_mail_list_delete(email: new_email)
+        end
+
+        it 'returns new email contact' do
+          contact = subject
```

Comment:
same here

Reply:
It would call subject for each thing I am testing.